File: 002/solution.py
from itertools import takewhile

def fibs():
    a, b = 0, 1
    while True:
        yield a+b
        a, b = b, a+b

even_fibs = (f for f in fibs() if not f%2) 
solution = sum(takewhile(lambda x: x < 4000000, even_fibs)) 

print(solution)

File: 002/solution.scala
object Solution {
  val fib: Stream[Long] = {
    def tail(h: Long, n: Long): Stream[Long] = h #:: tail(n, h+n)
    tail(0, 1)
  }
  def main(args: Array[String]) = {
    val solution = fib.takeWhile(_ < 4000000).toList.filter( _ % 2 == 0).sum
    println(solution)
  }
}
